server/src/main/java/org/opensearch/telemetry/tracing/handler/TraceableTransportResponseHandler.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing.handler;
+
+import org.opensearch.common.util.FeatureFlags;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.transport.TransportResponse;
+import org.opensearch.telemetry.tracing.Span;
+import org.opensearch.telemetry.tracing.SpanScope;
+import org.opensearch.telemetry.tracing.Tracer;
+import org.opensearch.transport.TransportException;
+import org.opensearch.transport.TransportResponseHandler;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Tracer wrapped {@link TransportResponseHandler}
+ * @param <T> TransportResponse
+ */
+public class TraceableTransportResponseHandler<T extends TransportResponse> implements TransportResponseHandler<T> {
+
+    private final Span span;
+    private final TransportResponseHandler<T> delegate;
+    private final Tracer tracer;
+
+    /**
+     * Constructor.
+     *
+     * @param delegate delegate
+     * @param span span
+     * @param tracer tracer
+     */
+    private TraceableTransportResponseHandler(TransportResponseHandler<T> delegate, Span span, Tracer tracer) {
+        this.delegate = Objects.requireNonNull(delegate);
+        this.span = Objects.requireNonNull(span);
+        this.tracer = Objects.requireNonNull(tracer);
+    }
+
+    /**
+     * Factory method.
+     * @param delegate delegate
+     * @param span span
+     * @param tracer tracer
+     * @return transportResponseHandler
+     */
+    public static TransportResponseHandler create(TransportResponseHandler delegate, Span span, Tracer tracer) {
```

Comment:
Code hygiene, please use generic types with generic paramaters:
```suggestion
        public static <S extends TransportResponse> TransportResponseHandler<?> create(TransportResponseHandler<S> delegate, Span span, Tracer tracer) {
{
if (FeatureFlags.isEnabled(FeatureFlags.TELEMETRY) == true) {
            return new TraceableTransportResponseHandler<S>(delegate, span, tracer);
        } else {
            return delegate;
        }
}
```       
```